File: SDKs/drchrono.com/v4 (Hunt Valley)/typescript/src/sdk/models/operations/usersread.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class UsersReadPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class UsersReadQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=doctor" })
  doctor?: number;
}


export class UsersReadSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  drchronoOauth2: shared.SchemeDrchronoOauth2;
}


export class UsersReadRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UsersReadPathParams;

  @SpeakeasyMetadata()
  queryParams: UsersReadQueryParams;

  @SpeakeasyMetadata()
  security: UsersReadSecurity;
}


export class UsersReadResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  userProfile?: shared.UserProfile;
}

File: SDKs/drchrono.com/v4 (Hunt Valley)/python/src/sdk/models/operations/users_read.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security
from ..shared import userprofile as shared_userprofile


@dataclasses.dataclass
class UsersReadPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class UsersReadQueryParams:
    doctor: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'doctor', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class UsersReadSecurity:
    drchrono_oauth2: shared_security.SchemeDrchronoOauth2 = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class UsersReadRequest:
    path_params: UsersReadPathParams = dataclasses.field()
    query_params: UsersReadQueryParams = dataclasses.field()
    security: UsersReadSecurity = dataclasses.field()
    

@dataclasses.dataclass
class UsersReadResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    user_profile: Optional[shared_userprofile.UserProfile] = dataclasses.field(default=None)
    
